### ISSUE TITLE
properties for zipWith

### ISSUE THREAD

**Issue by mechvel** [agda/agda-stdlib#228]
It looks like  Data.List.Properties misses these properties of zipWith :

```
zipWith-f-xs-[] :   ...   (f : A → B → C) → (xs : List A) → zipWith f xs [] ≡ []

length-zipWith :  ... →  (f : A → B → C) → (xs : List A) → (ys : List B) →
        	                    length (zipWith f xs ys) ≡ (length xs) ⊓ (length ys)
```

**Comment by gallais** [comment]
If I'm not mistaken there are no properties whatsoever about `zipWith`.
It would make sense to throw in some map-fusion properties too.

Probably also worth defining `unzipWith : (f : A -> B * C) -> List A -> List B * List C`
and proving `f . g == id -> zipWith f . unzipWith g = id`

**Comment by MatthewDaggitt** [comment]
I have a few of these lying around. I'll add them when I have a moment.

**Comment by MatthewDaggitt** [comment]
Closed with 5440369.

**Comment by gallais** [comment]
Nice, thanks! By map-fusion, I meant:

```agda
zipWith-map : zipWith f (map g xs) (map h ys) == zipWith (\ x y -> f (g x) (h y)) xs ys
map-zipWith : map g (zipWith f xs ys) == zipWith (\ x y -> g (f x y)) xs ys
```

**Comment by MatthewDaggitt** [comment]
Oh sorry, missed those. No I'll chuck them in myself.

**Comment by MatthewDaggitt** [comment]
Closed with 1da096b
